envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -218,12 +218,22 @@ class EnvoySnapshotFactory(
         globalSnapshot: GlobalSnapshot,
         egressRouteSpecifications: Collection<RouteSpecification>
     ): List<ClusterLoadAssignment> {
-        val egressRouteClusters = egressRouteSpecifications.map(RouteSpecification::clusterName)
+        val egressLoadAssignments = egressRouteSpecifications.mapNotNull { routeSpec ->
+            globalSnapshot.endpoints[routeSpec.clusterName]?.let { endpoints ->
+                // TODO: create a cache in GlobalSnapshot where a key is a pair (serviceName, serviceTag) and a value
+                //    is ClusterLoadAssignment (simple mutable map should be enough).
```

Comment:
+1 to idea of this cache

Reply:
I can do it in the next iteration. At the beginning it's not important I think, because not many Envoys will use this feature initially.